### PR TITLE
Verify paths

### PR DESCRIPTION
**Pull request by ilayzen**

### CODE REVIEW

File: pkg/plugin/oauth2/middleware.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ func RolesChecker(cache *cache.RolesCache, userRoles []string, path, method stri
 
 			for _, feature := range role.Features {
 				for _, endpoint := range feature.Endpoints {
-					if endpoint.Method == method && endpoint.Path == path {
+					if endpoint.Method == method && auditPaths(path, endpoint.Path) == true {
```
Comment (ribaraka) [inline pkg/plugin/oauth2/middleware.go]: `isEndpointPathsEqual`

File: pkg/plugin/oauth2/middleware.go [hunk 2]
```diff
@@ -134,3 +134,36 @@ func RolesChecker(cache *cache.RolesCache, userRoles []string, path, method stri
 
 	return fmt.Errorf("Error: Access is denied")
 }
+
+func auditPaths(endpointPath, userPath string) bool {
```
Comment (ribaraka) [inline pkg/plugin/oauth2/middleware.go]: isEndpointPathsEqual(a, b string)

File: pkg/plugin/oauth2/middleware.go [hunk 3]
```diff
@@ -134,3 +134,36 @@ func RolesChecker(cache *cache.RolesCache, userRoles []string, path, method stri
...
+	endpointPathArr := strings.Split(endpointPath, "/")
+	userPathArr := strings.Split(userPath, "/")
+	verifyPath := []string{}
+
+	for i, v := range endpointPathArr {
+		if v == userPathArr[i] {
```
Comment (ribaraka) [inline pkg/plugin/oauth2/middleware.go]: it might cause panic. pls handle this.

**Review by ribaraka — CHANGES_REQUESTED** [review]

File: pkg/plugin/oauth2/middleware.go (hunk 1 above)
Comment (ilayzen) [reply pkg/plugin/oauth2/middleware.go]: fixed

File: pkg/plugin/oauth2/middleware.go (hunk 2 above)
Comment (ilayzen) [reply pkg/plugin/oauth2/middleware.go]: I think its ban names for variables

**Review by ribaraka — APPROVED** [review]

**Pull request merged**